sharding/indexHashedNodesCoordinator.go

```diff
@@ -336,16 +334,13 @@ func (ihgs *indexHashedNodesCoordinator) GetValidatorWithPublicKey(
 		return nil, 0, ErrNilPubKey
 	}
 	ihgs.mutNodesConfig.RLock()
-	nodesConfig, ok := ihgs.nodesConfig[epoch]
-	ihgs.mutNodesConfig.RUnlock()
+	defer ihgs.mutNodesConfig.RUnlock()
```

Comment:
Add before the return from line 284: ihgs.mutNodesConfig.RUnlock()

Reply:
done